.packit.yaml

```diff
@@ -14,198 +14,142 @@ srpm_build_deps:
-                provisioning:
-                    tags:
-                        BusinessUnit: tmt
-    require:
-        label:
-          present:
-            - full test
-          absent:
-            - discuss
-
-  # Test internal wow
-  - job: tests
-    trigger: pull_request
-    identifier: "internal-wow"
-    targets:
-      - fedora-latest-stable
+      environments:
+        - &tmt-cloud-resources
+          settings:
+            provisioning:
+              tags:
+                BusinessUnit: tmt
+  - # Test internal wow
+    <<: *tests
     use_internal_tf: True
+    targets: *internal-targets
+    require: *require-full-tests
+    identifier: internal-wow
     fmf_url: "https://gitlab.cee.redhat.com/baseos-qe/integration_scripts.git"
     tmt_plan: "/tmt/integration/plan"
     tf_extra_params:
-        environments:
-          - settings:
-                provisioning:
-                    tags:
-                        BusinessUnit: tmt
-    require:
-        label:
-          present:
-            - full test
-          absent:
-            - discuss
+      environments:
+        - *tmt-cloud-resources
 
   # Build commits to main
-  - job: copr_build
+  - <<: *copr_build
     trigger: commit
     branch: main
-    targets:
-      - fedora-all
-      - epel-9
-    enable_net: False
     list_on_homepage: True
     preserve_project: True
     owner: "@teemtee"
     project: tmt
     release_suffix: "{PACKIT_PROJECT_BRANCH}"
     actions:
-        create-archive:
-          - hatch run docs:man
-          - hatch build -t sdist
-          - bash -c "ls dist/tmt-*.tar.gz"
-        get-current-version:
-          # Get rid of the the final version digit to make copr
-          # builds older than the final release
-          - bash -c "hatch version | sed -E 's/\\.[0-9]+\\.dev/.dev/'"
+      <<: *base-actions
+      get-current-version:
+        # Get rid of the the final version digit to make copr
+        # builds older than the final release
+        - bash -c "hatch version | sed -E 's/\\.[0-9]+\\.dev/.dev/'"
 
   # Release to copr
-  - job: copr_build
+  - <<: *copr_build
     trigger: release
-    targets:
-      - fedora-all
-      - epel-9
-    enable_net: False
     list_on_homepage: True
     preserve_project: True
     owner: "@teemtee"
     project: tmt
-    actions:
-        create-archive:
-          - hatch run docs:man
-          - hatch build -t sdist
-          - bash -c "ls dist/tmt-*.tar.gz"
-        get-current-version:
-          - hatch version
+    actions: *base-actions
 
   # Fedora releases
   - job: propose_downstream
     trigger: release
-    dist_git_branches:
-      - fedora-all
-      - epel-9
+    dist_git_branches: *all-targets
 
   - job: koji_build
     trigger: commit
-    allowed_pr_authors: ["packit", "psss", "lzachar"]
-    allowed_committers: ["packit", "psss", "lzachar"]
-    dist_git_branches:
-      - fedora-all
-      - epel-9
+    allowed_pr_authors: [ "packit", "psss", "lzachar" ]
```

Comment:
looking at this, I am wondering if we should revisit the hardcoded two members here.

1. First of all, why do we need this?

https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters

2. if we really do, I would rather ask for a `tmt` group in Fedora to control who has this access

Reply:
It's a double-edged sword because it can mean that you can't build in a side-tag without triggering the main build. There should be a label reading soon in packit to disable the build if something like `no-build` label is attached to the PR. For that I prefer to have only `packit` on both `allowed_pr_authors` and `allowed_committers`. If packit opens the PRs, all should be fine in the current workflow.

Reply:
Just checked the config documentation. I guess we could use `all_committers` as the value here? That would allow everybody who has a commit access to the dist git repo auto-create koji builds.

> It's a double-edged sword because it can mean that you can't build in a side-tag without triggering the main build.

I'm not sure I understand here. The `allowed_pr_authors` is used to allow auto-trigerring koji build when the dist git pull request is merged. Do we (will we) need side-tags? Am I missing something?

Sometimes a manual action / adjustment of the dist git pull request is needed so I'd say this setting might come handy.

Reply:
The idea is that `packit` does not support *not* triggering an automatic koji build when a manual action is performed (haven't checked in a while). On the other hand, if we are *manually* performing a git push, why shouldn't we be doing the `koji_build` manually as well? For the PR case it's 50-50 if we want it or not, given that packit could eventually support skipping `koji_build` depending on the presence of a `label`/`tag` in the PR. For the PR case there would be a way to override this behavior currently by pushing the PR commit manually.

Thinking about this more, I think at the very least `allowed_committers` should be removed.

Reply:
> On the other hand, if we are _manually_ performing a git push, why shouldn't we be doing the `koji_build` manually as well?

I remember recently there was a minor bump on the fedora release road and it was necessary to add a commit to the pagure pull request. Everything else then can go on in the usual / automated way.

Let's cover this change in a separate pull request not to block merging this one.